sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/test-aot-compat.ps1

```diff
@@ -0,0 +1,35 @@
+param([string]$targetNetFramework)
+
+dotnet restore
+$publishOutput = dotnet publish --framework net7.0 -nodeReuse:false /p:UseSharedCompilation=false /p:ExposeExperimentalFeatures=true
+
+if ($LASTEXITCODE -ne 0)
+{
+    Write-Host "Publish failed."
+    Write-Host $publishOutput
+    Exit 2
+}
+
+$actualWarningCount = 0
+
+foreach ($line in $($publishOutput -split "`r`n"))
+{
+    if ($line -like "*analysis warning IL*")
+    {
+        Write-Host $line
+
+        $actualWarningCount += 1
+    }
+}
+
+Write-Host "Actual warning count is:", $actualWarningCount
+$expectedWarningCount = 7
```

Comment:
Might be worth adding a comment what are the warnings we're expecting here - hopefully this will get close to zero soon, but still.

Reply:
done. 
yeah @m-redding already has a PR addressing half of the remaining issues.